clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17534,79 +17534,6 @@ Decl *Sema::BuildStaticAssertDeclaration(SourceLocation StaticAssertLoc,
   return Decl;
 }
 
-/// Perform semantic analysis of the given friend type declaration.
-///
-/// \returns A friend declaration that.
-FriendDecl *Sema::CheckFriendTypeDecl(SourceLocation LocStart,
```

Comment:
Why remove the function? 
I think it might be cleaner to add additional parameters to it ( ie, the template arguments)

Reply:
@cor3ntin We call `CheckFriendTypeDecl` from two places: `Sema::ActOnFriendTypeDecl`, and `TemplateDeclInsantiator::VisitFriendDecl`. However, the only thing that `CheckFriendTypeDecl` does during instantiation is call `FriendDecl::Create` -- the other checks are only done when initially parsing the declaration.